libpod/image/image.go

```diff
@@ -567,3 +567,63 @@ func Import(path, reference string, writer io.Writer, signingOptions SigningOpti
 	}
 	return runtime.NewFromLocal(reference)
 }
+
+// MatchRepoTag takes a string and tries to match it against an
+// image's repotags
+func (i *Image) MatchRepoTag(input string) (string, error) {
+	var maxKey string
+	// first check if we have an exact match with the input
+	if util.StringInSlice(input, i.Names()) {
+		return input, nil
+	}
+
+	results := make(map[string]int)
+
+	// next check if we are missing the tag
+	dcImage, err := decompose(input)
+	if err != nil {
+		return "", err
+	}
+
+	for _, repoName := range i.Names() {
```

Comment:
I'd like to get some review from people more familiar with image code than myself on this matching algorithm. I feel like there should be a better way, but I can't think of one myself.

Reply:
i asked around, there isnt ...